opa/opa.go

```diff
@@ -24,6 +25,27 @@ const (
 	xForwardedAccessTokenHeader = "X-Forwarded-Access-Token" //nolint:gosec
 )
 
+//nolint:gochecknoglobals
+// regoFunctions map is used for the providers' self-registration.
+var regoFunctions map[string]func(log.Logger) func(*rego.Rego)
+
+//nolint:gochecknoglobals
+// regoFunctionsMtx is used to protect the providerFactories.
+var regoFunctionsMtx sync.RWMutex
+
+//nolint:gochecknoinits
+func init() {
+	regoFunctions = make(map[string]func(log.Logger) func(*rego.Rego))
+}
+
+// onboardNewFunction is used by the ...
```

Comment:
Unfinished comment perhaps?

Reply:
Ooops, that is a leftover from the initial PoC, I will update it.